fabric-events-player-v1/src/main/java/net/fabricmc/mixin/event/player/MixinPlayerManager.java

```diff
@@ -0,0 +1,19 @@
+package net.fabricmc.mixin.event.player;
+
+import net.fabricmc.api.event.v1.player.PlayerRespawnCallback;
+import net.minecraft.server.PlayerManager;
+import net.minecraft.server.network.ServerPlayerEntity;
+import net.minecraft.world.dimension.DimensionType;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+@Mixin(PlayerManager.class)
+public abstract class MixinPlayerManager {
+	@Inject(at = @At("TAIL"), method = "respawnPlayer(Lnet/minecraft/server/network/ServerPlayerEntity;Lnet/minecraft/world/dimension/DimensionType;Z)Lnet/minecraft/server/network/ServerPlayerEntity;")
```

Comment:
I think the benefits of this mixin target needs to be documented. A standard approach is to hook `ServerPlayerEntity#copyFrom`, as this is where the actual player data is copied. However, targeting the tail of `respawnPlayer` means callbacks are able to reposition the respawning player, or change its dimension.

Reply:
So like a comment justifying why I've chosen this injection rather than copyFrom

Reply:
Also note in the javadoc that respawning players *can* be repositioned.